File: backend/app/models/chamado.py
from sqlalchemy import Column, Integer, String, Text, Enum, ForeignKey
from sqlalchemy.orm import relationship

from database import Base
from app.enums.chamados import StatusChamado, PrioridadeChamado


class Chamado(Base):
    __tablename__ = "chamados"

    id = Column(Integer, primary_key=True)
    titulo = Column(String(100), nullable=False)
    descricao = Column(Text, nullable=False)
    status = Column(
        Enum(
            StatusChamado,
            values_callable=lambda enum: [item.value for item in enum]
        ),
        nullable=False
    )

    prioridade = Column(
        Enum(
            PrioridadeChamado,
            values_callable=lambda enum: [item.value for item in enum]
        ),
        nullable=False
    )
    solicitante_id = Column(
        Integer,
        ForeignKey("usuarios.id"),
        nullable=False,
    )
    solicitante = relationship(
    "Usuario",
    back_populates="chamados"
    )

    historicos = relationship(
        "HistoricoChamado",
        back_populates="chamado",
        cascade="all, delete-orphan"
    )

File: frontend/src/app/shared/models/chamado.ts
export type StatusChamado =
    | 'Novo'
    | 'Pendente'
    | 'Em andamento'
    | 'Encerrado'
    | 'Cancelado';

export type PrioridadeChamado =
    | 'P1 - Muito alta'
    | 'P2 - Alta'
    | 'P3 - Média'
    | 'P4 - Baixa'
    | 'P5 - Muito baixa';

export interface Chamado {
    id: number;
    titulo: string;
    descricao: string;
    status: StatusChamado;
    prioridade: PrioridadeChamado;
    solicitante_id: number;
}